### DISCUSSION TITLE
Curious why Seq isn't supporting non-TLS gRPC ingestion!

### DISCUSSION THREAD

**Discussion by gkinsman** [datalust/seq-tickets#1845] [General]
Hi there! The OTel stuff is super cool, looking forward to seeing it progress! I'm curious, though sure you've considered, why not support gRPC ingestion over non-HTTPS endpoints? It's possible to configure Kestrel to support HTTP2 over non-TLS by explicitly setting the protocol to HTTP2, though perhaps that would require having a third non-TLS, HTTP2-only endpoint alongside the HTTP1 and HTTPS endpoints? That would be a bit of a confusing pain to be fair.

Thanks!

**Comment by nblumhardt** [comment]
Hi @gkinsman, great to hear from you.

Our plan right now is to sidestep this for OTLP and just implement the `http/proto` spec option for use in non-TLS scenarios. We're set to merge, build, and hopefully post a new preview that includes it later today.

We've considered options along the lines of what you're describing, but reached the conclusion that, for now, it'd be too awkward for us to explain and support.

"OpenTelemetry over gRPC requires HTTPS, otherwise use plain HTTP/proto" is the soundbite that sums up our approach. Keen to know what you think!

**Comment by gkinsman** [comment]
Makes total sense and likely avoids the confusion of 'I thought gRPC requires HTTPS' 👍. Thanks for the explanation! Can't wait to use Seq for local OTel Dev ☺️
